File: backend/src/main/java/com/nishtha/RoomEase/complaint/service/ComplaintService.java
package com.nishtha.RoomEase.complaint.service;


import com.nishtha.RoomEase.complaint.dto.ComplaintResponse;
import com.nishtha.RoomEase.complaint.dto.CreateComplaintRequest;
import com.nishtha.RoomEase.complaint.dto.UpdateComplaintRequest;

import java.util.List;

public interface ComplaintService {

    ComplaintResponse createComplaint(CreateComplaintRequest request);

    List<ComplaintResponse> getMyComplaints();

    List<ComplaintResponse> getComplaintsForProperty(Long propertyId);

    ComplaintResponse updateComplaint(Long complaintId, UpdateComplaintRequest request);
}

File: frontend/src/services/complaintService.js
import api from "./api";

const complaintService = {

    createComplaint: async (data) => {
        const response = await api.post("/complaints",data);
        return response.data;
    },


    getMyComplaints: async () => {
        const response = await api.get("/complaints/my");
        return response.data;
    },


    getPropertyComplaints: async (propertyId) => {

        const response = await api.get(`/complaints/property/${propertyId}`);
        return response.data;
    },


    updateComplaint: async (complaintId, data) => {
        const response = await api.patch(`/complaints/${complaintId}`,data);
        return response.data;
    }

};

export default complaintService;